### PR TITLE
Update browser.js to use apiv4 charmstore api

### PR DESCRIPTION
**Pull request by hatched**

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

Visit 
localhost:8888/?deploy-target=asdf/ghost
which should make a notification that there is an error
Visit
localhost:8888/?deploy-target=precise/ghost
and it should deploy as expected

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2429/
Test FAILed.

File: app/subapps/browser/browser.js [hunk 1]
```diff
@@ -232,26 +230,40 @@ YUI.add('subapp-browser', function(Y) {
...
+                  undefined, //config file content
+                  1, // number of units
+                  {}, //constraints
+                  null); // toMachine
+              this.fire('autoplaceAndCommitAll');
+            }.bind(this),
+            function(error) {
+              var message = 'Unable to deploy target: ' + entityId;
+              try {
+                message = JSON.parse(error.currentTarget.responseText).Message;
```
Comment (makyo) [inline app/subapps/browser/browser.js]: Non-actionable - We've been using Y.JSON in the past; should we just default to the browser's parser in the future?

**Comment by makyo** [comment]
:+1: on code, QAing now.

**Comment by huwshimi** [comment]
Looks good, thanks @hatched :+1: No QA.

**Comment by makyo** [comment]
QA okay, and bundles still deploy as expected.  For a future branch, deploying a requested bundle that returns a 404 does not show an error notification (e.g: http://localhost:8888/?deploy-target=bundle/mediawiki-simple ), had to check the console to see the 404.

File: app/subapps/browser/browser.js (hunk 1 above)
Comment (hatched) [inline app/subapps/browser/browser.js]: Yeah I'd say so - the Y.JSON wrapper simply checks for the natives existence then passes through to it so this is saving a function call. Also all of our supported browsers have `JSON`

**Comment by hatched** [comment]
Thanks for the reviews and QA!

@makyo that was a good catch and the fix has been implemented 

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2430/
Test PASSed.

**Pull request merged**